### DISCUSSION TITLE
Support custom charset and collation for schema_migrations table (mysql)

### DISCUSSION THREAD

**Discussion by mheffner** [amacneil/dbmate#390] [Feature Ideas]
For MySQL the default `schema_migrations` table is created using a latin1 charset and latin1_bin collation:
```
> show create table schema_migrations;
+-------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table             | Create Table                                                                                                                                              |
+-------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------+
| schema_migrations | CREATE TABLE `schema_migrations` (
  `version` varchar(255) NOT NULL,
  PRIMARY KEY (`version`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1 COLLATE=latin1_bin |
+-------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------+
```
This is incompatible with some database providers, like Planetscale. Planetscale requires a utf8 charset/collation.

It looks like this is hardcoded [here](https://github.com/amacneil/dbmate/blob/9f1f34602cb010278fbc3c6a69fe9faa3fb7db9b/pkg/driver/mysql/mysql.go#L249). It would be great if this were configurable.

**Comment by amacneil** [comment]
Some history on this:
- https://github.com/amacneil/dbmate/issues/85
- https://github.com/amacneil/dbmate/pull/172

Potential solutions:
- Change the version varchar length to 128 and remove the explicit charset
- Make it configurable (how)

Thoughts? I'm leaning towards the first option.

**Reply by mheffner** [reply]
Thanks for the context! Yes, the first option makes the most sense IMO given your [comment](https://github.com/amacneil/dbmate/issues/85#issuecomment-534197224) also mentions that these fields are usually only 14 chars long.

**Reply by amacneil** [reply]
PR: https://github.com/amacneil/dbmate/pull/391
